client/rest/src/main/java/org/elasticsearch/client/RequestOptions.java

```diff
@@ -71,6 +71,13 @@ public Builder toBuilder() {
         return headers;
     }
 
+    /**
+     * Return true if the options contain the given header
+     */
+    public boolean containsHeader(String name) {
+        return headers.stream().anyMatch(h -> name.equals(h.getName()));
```

Comment:
HTTP headers are [case-insentitive](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2). We should use `equalsIgnoreCase` here and in `removeHeader()`.